src/dmd/hdrgen.d

```diff
@@ -1606,6 +1606,16 @@ public:
         if (!f.inferRetType && tf.next)
             typeToBuffer(tf.next, null, buf, hgs);
         parametersToBuffer(tf.parameterList, buf, hgs);
+
+        // https://issues.dlang.org/show_bug.cgi?id=20074
+        void printAttribute(string str)
+        {
+            buf.writeByte(' ');
+            buf.writestring(str);
+        }
+        tf.attributesApply(&printAttribute);
```

Comment:
The comment three lines above the patch says that omitting the attributes is deliberate. It actually can be a bit annoying in the error message if the problem is not related to inferred attributes.

Reply:
I agree, but on the other hand it may lead to (falsely) uncompilable code. Consider:

```d
class C
{
    this() @safe
    {
        () @trusted
        {
            int a; int*p = &a;                                                                                                                
        }();
    }   
}
```

Without this patch, the compiler will omit `@trusted` and generate uncompilable code from perfectly fine code.

The comment baffled me also, but I think that this is the correct way to go.

Reply:
You should adjust or remove the comment in that case. A compromise might be to emit explicitly specified attributes, but not inferred ones. But the required info might not exist and not worth the complications to add.